src/bin.js

```diff
@@ -55,20 +59,36 @@ async function main() {
   if (gExecutable) {
     // Update the path so that the replay version of node is found first when
     // subprocesses run.
-    process.env.PATH = `${getDirectory()}/node:` + process.env.PATH;
+    process.env.PATH = `${getNodeDirectory()}:` + process.env.PATH;
 
     const rv = spawnSync(argv[0], argv.slice(1), { stdio: "inherit" });
     process.exit(rv.status);
   }
 
-  const rv = spawnSync(`${getDirectory()}/node/node`, argv, { stdio: "inherit" });
+  const rv = spawnSync(`${getNodeDirectory()}/node`, argv, {
+    stdio: "inherit",
+  });
   process.exit(rv.status);
 }
 
 async function updateNode() {
   if (!fs.existsSync(getDirectory())) {
     fs.mkdirSync(getDirectory());
   }
+
+  if (process.env.RECORD_REPLAY_NODE_DIRECTORY) {
+    if (!fs.existsSync(`${process.env.RECORD_REPLAY_NODE_DIRECTORY}/node`)) {
+      throw new Error(
+        `Environment variable for a manually managed "RECORD_REPLAY_NODE_DIRECTORY" was specified, but no "node" binary could be found there. Exiting`
+      );
+    }
+    if (gNeedUpdate) {
```

Comment:
Since this is only `true` when explicitly when the user adds the `--update` flag, I'd have expected us to apply the update. Was this added as a NixOS compatibility consideration?

Reply:
In NixOS it would just fail because it's a readonly file system - so leaving it out would be okay as well. 

My reasoning was more that I was not exactly sure where to put/search for the JSON file in the case we define a custom "node binary path" - and that would assume that such a JSON file would be there in the first place, which adds another layer of complexity for people using that environment variable.

And lastly, as far as I understand it, the similar environment variable `RECORD_REPLAY_DRIVER` also disables automatic updates.

So it just seemed like opening a can of worms that would add lots of complexity - when actually most users who want to use this feature probably want to opt out of said complexity.

Reply:
I'm not the expert on this code but I think the expectation is that `replay-node` can both install and update the version of node. So if the user wants to specify the install path of node (maybe for CI ... maybe because they're particular), they should be able to do that with the env variable.

IOW, if they install this and run `RECORD_REPLAY_NODE_DIRECTORY=~/.my-node-path replay-node my-script` it should download node to that directory and then run the script. I think this flow makes sense and generally matches how it works now with the default directory.

I don't _think_ setting `RECORD_REPLAY_DRIVER` will disable installing or updating `node`. The driver is usually linked at build time but can be dynamically linked with that variable (usually for dev testing).

Regarding the JSON file, normally that's placed in the same directory as the node binary. Given how the update logic is written now, I think this makes sense since it's really just checking to see if the latest version of that file is the same as the local one for this node install. This means that a user _could_ install multiple copies of `replay-node` and each would manage its own update.

Reply:
Hmm, that's valid as well.

Suggestion:
on `--update`,
* if neither `node` nor the json file are present, install `node` into the target path and save the json
* if both `node` and the json file are present, update if necessary
* if only `node`, but not the json file are present, throw an exception explaining that the update functionality requires the JSON to be present as well.

alternatively:

* if only `node`, but not the json file are present, download the newest `node` and json

Personally I think the "only update if both or neither are present" is the most predictable outcome, but I'll change this to whatever you decide makes most sense :)

Reply:
Sorry I lost track of this. I think your suggestion makes sense. The exception should give enough context to anyone running this command on how to proceed and sounds like it'll be a good fit for the NixOS scenario. Thanks!